### ISSUE TITLE
[Feature Request] Exporting from command line

### ISSUE THREAD

**Issue by wendelscardua** [BleuBleu/FamiStudio#26]
This is not the most important feature ever, but it would be nice to be able to export songs from command line.

Something like `FamiStudio.exe my-project.fms -export my-project.s -type FamiTone2 -format ca65 -no-separate-files`

**Comment by BleuBleu** [comment]
Hey hi!

Funny enough, this was added last week. It will come out in 2.1.0, which is still probably 3 weeks away.

This is still subject to change, but the syntax so far looks like this. Its a bit verbose but i wanted to support every input and output format.

> FamiStudio 2.0.1 Command-Line Usage
> 
> Usage:
>   FamiStudio <input> <command> <output> [-options]
> 
> Examples:
>   FamiStudio music.fms wav-export music.wave -export-songs:2 -wav-export-rate:48000
>   FamiStudio music.fms famitracker-txt-export music.txt -export-songs:0,1,2
>   FamiStudio music.fms famitone2-export music.s -famitone2-format:ca65
> 
> Supported input formats:
>   FamiStudio project (*.fms)
>   FamiStudio text file (*.txt)
>   FamiTracker 0.4.6 file (*.ftm)
>   FamiTracker 0.4.6 text file (*.txt)
>   Nintendo Sound Format (*.nsf, *.nsfe)
> 
> Supported commands and corresponding output format(s):
>   wav-export : Export to a WAV file (*.wav).
>   nsf-export : Export to a NSF file (*.nsf).
>   rom-export : Export to a NES ROM file (.nes).
>   famitracker-txt-export : Export to a FamiTracker text file (*.txt).
>   famistudio-txt-export : Export to a FamiStudio text file (*.txt).
>   famitone2-export : Export to FamiTone2 assembly file(s) (*.s, *.asm).
> 
> General options:
>   -export-songs:<songs> : Comma-seperated zero-based indices of the songs to export (default:all).
> 
> NSF import specific options
>   -nsf-import-song:<song> : Zero-based index of the song to import (default:0).
>   -nsf-import-duration:<duration> : Duration, in sec, to record from the NSF (default:120).
>   -nsf-import-pattern-length:<length> : Pattern length to split the NSF into (default:256).
>   -nsf-import-start-frame:<frame> : Frame to skips before starting the NSF capture (default:0).
> 
> WAV export specific options
>   -wav-export-rate:<rate> : Sample rate of the exported wave : 11025, 22050, 44100 or 48000 (default:44100).
> 
> NSF export specific options
>   -nsf-export-mode:<mode> : Target machine: ntsc, pal or dual (default:ntsc).
> 
> FamiTone2 export specific options
>   -famitone2-format:<format> : Assembly format to export to : nesasm, ca65 or asm6 (default:nesasm).
>   -famitone2-variant:<variant> : The variant of FamiTone2 to use : famitone2, famitone2fs or famistudio (default:famitone2).
>   -famitone2-seperate-files : Export songs to individual files, output filename is the output path (default:disabled).
>   -famitone2-seperate-song-pattern:<pattern> : Name pattern to use when exporting songs to seperate files (default:{project}_{song}).
>   -famitone2-seperate-dmc-pattern:<pattern> : DMC filename pattern to use when exporting songs to seperate files (default:{project}).
>

**Comment by wendelscardua** [comment]
Well, if it's already done I'll close the issue.

Thanks :)
